src/entities/trades/curve/CurveTrade.ts

```diff
@@ -230,6 +224,9 @@ export class CurveTrade extends Trade {
 
     // Use Custom contract for native xDAI<>USDT and xDAI<>USDC trades on Gnosis Chain
     if (chainId === ChainId.XDAI && (isNativeAssetIn || isNativeAssetOut)) {
+      if (isNativeAssetIn) {
+        value = amountInBN.toString()
+      }
```

Comment:
This condition here rules out trading ETH on Ethereum and other networks. The stETH, for example, allows trading between ETH and stETH.

Reply:
yeah we fixed that!